sdk/keyvault/keyvault-certificates/src/index.ts

```diff
@@ -1551,28 +1693,82 @@ export class CertificateClient {
   ): DeletedCertificate {
     const parsedId = parseKeyvaultEntityIdentifier("certificates", certificateBundle.id);
 
+    const attributes: any = certificateBundle.attributes || {};
+    const policy = toPublicPolicy(certificateBundle.policy || {});
+
+    let abstractProperties: any = {
+      keyId: certificateBundle.kid,
+      secretId: certificateBundle.sid,
+      name: parsedId.name,
+      createdOn: attributes.created,
+      updatedOn: attributes.updated,
+      expiresOn: attributes.expires,
+      ...parsedId,
+      ...certificateBundle,
+      ...certificateBundle.attributes
+    };
+
+    if (abstractProperties.deletedDate) {
+      delete abstractProperties.deletedDate;
+    }
+
+    if (abstractProperties.expires) {
+      delete abstractProperties.expires;
+    }
+    if (abstractProperties.created) {
+      delete abstractProperties.created;
+    }
+    if (abstractProperties.updated) {
+      delete abstractProperties.updated;
+    }
+
+    delete abstractProperties.policy;
+
     return {
-      contentType: certificateBundle.contentType as CertificateContentType,
+      keyId: certificateBundle.kid,
+      secretId: certificateBundle.sid,
+      name: parsedId.name,
       recoveryId: certificateBundle.recoveryId,
       scheduledPurgeDate: certificateBundle.scheduledPurgeDate,
-      deletedDate: certificateBundle.deletedDate,
-      properties: {
-        ...parsedId,
-        ...certificateBundle,
-        ...certificateBundle.attributes
-      }
+      deletedOn: certificateBundle.deletedDate,
+      contentType: certificateBundle.contentType as CertificateContentType,
+      policy,
+      properties: abstractProperties
```

Comment:
The exact same set of code is being added to both `getCertificateFromCertificateBundle` and `getDeletedCertificateFromDeletedCertificateBundle`.

Please consider updating `getDeletedCertificateFromDeletedCertificateBundle()` as follows
```javascript
const certificate = getCertificateFromCertificateBundle(certificateBundle);
certificate.recoveryId = certificateBundle.recoveryId;
certificate.scheduledPurgeDate = certificateBundle.scheduledPurgeDate;
certificate.deletedOn = certificateBundle.deletedOn;
```

Reply:
This has been handled in @jonathandturner's commit https://github.com/Azure/azure-sdk-for-js/pull/5991/commits/1ef22d22baf363462e791bd9064546c1208a0bdf